### PR TITLE
lock: update target os-type and os-version after provisioning

### PR DESCRIPTION
**Pull request by ryg-**
https://tracker.ceph.com/issues/42081

### CODE REVIEW

**Comment by kshtsk** [comment]
jenkins test

File: teuthology/lock/cli.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import teuthology.provision
 from teuthology import misc
 from teuthology.config import set_config_attr
+from teuthology.config import config
```
Comment (kshtsk) [inline teuthology/lock/cli.py]: why not `from teuthology import config`

File: teuthology/lock/cli.py [hunk 2]
```diff
@@ -149,7 +150,11 @@ def main(ctx):
                 ctx.machine_type, ctx.os_type, ctx.os_version):
             log.error('Invalid os-type or version detected -- lock failed')
             return 1
-        reimage_types = teuthology.provision.fog.get_types()
+        reimage_types = None
+        if config.get('provision_system', 'FOG') == 'pelagos':
+            reimage_types = teuthology.provision.pelagos.get_types()
+        else:
+            reimage_types = teuthology.provision.fog.get_types()
```
Comment (kshtsk) [inline teuthology/lock/cli.py]: I think this code is not related to this PR

File: teuthology/lock/ops.py [hunk 3]
```diff
@@ -80,7 +90,11 @@ def lock_many(ctx, num, machine_type, user=None, description=None,
         # Only query for os_type/os_version if non-vps and non-libcloud, since
         # in that case we just create them.
         vm_types = ['vps'] + teuthology.provision.cloud.get_types()
-        reimage_types = teuthology.provision.fog.get_types()
+        reimage_types = None
+        if config.get('provision_system', 'FOG') == 'pelagos':
+            reimage_types = teuthology.provision.pelagos.get_types()
+        else:
+            reimage_types = teuthology.provision.fog.get_types()
```
Comment (kshtsk) [inline teuthology/lock/ops.py]: I think this code is not related to this PR

File: teuthology/lock/cli.py (hunk 1 above)
Comment (ryg-) [reply teuthology/lock/cli.py]: fixed

File: teuthology/lock/cli.py (hunk 2 above)
Comment (ryg-) [reply teuthology/lock/cli.py]: removed

File: teuthology/lock/ops.py (hunk 3 above)
Comment (ryg-) [reply teuthology/lock/ops.py]: removed

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
@gregsfortytwo can we test this?

**Comment by gregsfortytwo** [comment]
ok to test

**Comment by kshtsk** [comment]
I agree with @gregsfortytwo 
However it requires at least to run on time in the lab? Can someone poke Zack to look at it?

**Comment by gregsfortytwo** [comment]
I'm having trouble parsing why but something went wrong (involving flake8?).

**Comment by gregsfortytwo** [comment]
@kshtsk I think you can just check out this branch anywhere that has sepia access and do the lock ops, was there something else you're worried about?

**Comment by kshtsk** [comment]
> I'm having trouble parsing why but something went wrong (involving flake8?).

Yeah, it is masqueraded 
```
teuthology/lock/cli.py:14:1: F401 'teuthology.config' imported but unused
ERROR: InvocationError for command /home/jenkins-build/build/workspace/teuthology-pull-requests/.tox/flake8/bin/flake8 --select=F,E9 teuthology scripts (exited with code 1)
```

**Comment by ryg-** [comment]
Import was removed in last update
> Yeah, it is masqueraded
> 
> ```
> teuthology/lock/cli.py:14:1: F401 'teuthology.config' imported but unused

File: teuthology/lock/ops.py [hunk 4]
```diff
@@ -11,13 +11,31 @@
...
+import teuthology.task.internal
+from teuthology.misc import canonicalize_hostname
 
 import util
 import keys
 
 log = logging.getLogger(__name__)
 
 
+def update_nodes(nodes, reset_os=False):
```
Comment (kshtsk) [inline teuthology/lock/ops.py]: Just wondering if we can join this method with update_inventory, like:
```
def update_inventory(nodes, reset=True):
  if isinstance(nodes, list):
    # process list as names of nodes
    .
    .
    .
  if isinstance(nodes, dict):
    # process dict
    nodes = [nodes]
    .
    .
    .
```

**Comment by djgalloway** [comment]
```
2019-10-09 15:07:26,050.050 INFO:teuthology.orchestra.run.smithi080:Running:
2019-10-09 15:07:26,050.050 INFO:teuthology.orchestra.run.smithi080:> hostname
2019-10-09 15:07:26,502.502 INFO:teuthology.orchestra.run.smithi080.stdout:smithi080.front.sepia.ceph.com
2019-10-09 15:07:26,503.503 INFO:teuthology.orchestra.run.smithi080:Running:
2019-10-09 15:07:26,503.503 INFO:teuthology.orchestra.run.smithi080:> grep smithi080.front.sepia.ceph.com /etc/hosts
2019-10-09 15:07:26,631.631 INFO:teuthology.orchestra.run.smithi080.stdout:172.21.15.80 smithi080.front.sepia.ceph.com smithi080
2019-10-09 15:07:26,631.631 INFO:teuthology.orchestra.run.smithi080:Running:
2019-10-09 15:07:26,631.631 INFO:teuthology.orchestra.run.smithi080:> sudo hostname smithi080
2019-10-09 15:07:26,802.802 INFO:teuthology.orchestra.run.smithi080:Running:
2019-10-09 15:07:26,802.802 INFO:teuthology.orchestra.run.smithi080:> sudo sed -i -e 's/smithi080.front.sepia.ceph.com/smithi080/g' /etc/hosts
2019-10-09 15:07:26,942.942 INFO:teuthology.orchestra.run.smithi080:Running:
2019-10-09 15:07:26,942.942 INFO:teuthology.orchestra.run.smithi080:> sudo sed -i -e 's/172.21.15.80/172.21.15.80/g' /etc/hosts
2019-10-09 15:07:27,082.082 INFO:teuthology.orchestra.run.smithi080:Running:
2019-10-09 15:07:27,082.082 INFO:teuthology.orchestra.run.smithi080:> sudo sed -i -e 's/smithi080.front.sepia.ceph.com/smithi080/g' /etc/hostname
2019-10-09 15:07:27,224.224 INFO:teuthology.orchestra.run.smithi080:Running:
2019-10-09 15:07:27,224.224 INFO:teuthology.orchestra.run.smithi080:> sudo hostname smithi080
2019-10-09 15:07:27,359.359 INFO:teuthology.provision.fog.smithi080:Deploy complete!
2019-10-09 15:07:28,247.247 INFO:teuthology.lock.keys:Checking smithi080.front.sepia.ceph.com
2019-10-09 15:07:28,249.249 INFO:teuthology.lock.keys:New key found. Updating...
2019-10-09 15:07:28,323.323 INFO:teuthology.lock.ops:Updating [smithi080.front.sepia.ceph.com]: set os type and version on server
```

File: teuthology/lock/ops.py (hunk 4 above)
Comment (kshtsk) [reply teuthology/lock/ops.py]: @ryg- may be you can refactor this patch later.

File: teuthology/lock/ops.py [hunk 5]
```diff
@@ -101,6 +119,7 @@ def lock_many(ctx, num, machine_type, user=None, description=None,
                 machines=', '.join(machines.keys())))
             if machine_type in vm_types:
                 ok_machs = {}
+                update_nodes(machines, True)
```
Comment (kshtsk) [inline teuthology/lock/ops.py]: it is not clear in context what is 'True', so better to right `reset=True`.

File: teuthology/lock/ops.py [hunk 6]
```diff
@@ -11,13 +11,31 @@
...
+        remote = teuthology.orchestra.remote.Remote(
+            canonicalize_hostname(node))
+        if reset_os:
+            log.info("Updating [%s]: reset os type and version on server", node)
+            inventory_info = dict()
+            inventory_info['os_type'] = ''
+            inventory_info['os_version'] = ''
+            inventory_info['name'] = remote.hostname
+        else:
+            log.info("Updating [%s]: set os type and version on server", node)
```
Comment (kshtsk) [inline teuthology/lock/ops.py]: Please, remove [] around %s, it is not style in of this code base. Also it is better to add actual values of os_type and os_version to this log info.

File: teuthology/lock/ops.py [hunk 7]
```diff
@@ -11,13 +11,31 @@
...
 
 log = logging.getLogger(__name__)
 
 
+def update_nodes(nodes, reset_os=False):
+    for node in nodes:
+        remote = teuthology.orchestra.remote.Remote(
+            canonicalize_hostname(node))
+        if reset_os:
+            log.info("Updating [%s]: reset os type and version on server", node)
```
Comment (kshtsk) [inline teuthology/lock/ops.py]: nit: 
```
log.info("Reset os type and version for machine %s", node)
```
